src/Pages/ProductList/ProductList.jsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import "./ProductList.css";
+import {products} from "../../Assets/Data"
+import ProductCard from "../../Components/ProductCard/ProductCard";
+import Filter from "../../Components/Filter/Filter";
+import { useFilter } from "../../context/filterContext";
+
+import { sortProduct } from "../../Utils/Sort";
+import { categoryFilter } from "../../Utils/Category";
+import { PriceFilter } from "../../Utils/PriceFilter";
+import { BestSellerFunc } from "../../Utils/BestSeller";
+
+export const ProductList = () => {
+  const {
+    state: { sortBy, category, price, BestSeller  },
+  } = useFilter();
+
+  
+  const sortedProducts = sortProduct(products, sortBy);
+  const categoryFilteredProducts = categoryFilter(sortedProducts,category)
+  const bookBestSeller = BestSellerFunc(categoryFilteredProducts, BestSeller)
+  const priceFilteredProducts = PriceFilter(bookBestSeller,price)
```

Comment:
Here you can use the concept of composition of functions and use it here
For example: [See this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#function_composition_enabling_piping)